### ISSUE TITLE
Disable "Expand Sync to Working Set" by default

### ISSUE THREAD

**Issue by fkorotkov** [bazelbuild/intellij#3215]
I want to bring to attention one scenario that I experiences which confused me and my investigation into it.

**TLDR:** I propose to change the default for [`BlazeUserSettings#expandSyncToWorkingSet `](https://github.com/bazelbuild/intellij/blob/4c91d565062e152cace081bf3bfc8860008dc145/base/src/com/google/idea/blaze/base/settings/BlazeUserSettings.java#L72) from `true` to `false` because IMO it can confuse users more then help them.

In my case I have a huge mono repository where we practice fork driven approach (each engineer has a fork to work on and we submit PRs to the main repository). I have two origins in my Git repository: `origin` and `fork`. For example, I have a feature branch  `fork/fedor/feature-1` which was checkout from `fork/master`. I did some development, opened a PR, got some feedback after a few days and now I'm rebasing my feature branch on top of the latest changes from `origin/master`. Now when I run Sync from IntelliJ I see that the plugin **syncs every single target in the repository**.

After some investigation I found the root cause: because my `fork/master` hasn't been rebased as well this causes the "working set" to contain all files changed between `fork/fedor/feature-1` and the outdated `fork/master` instead of `origin/master`. [Here](https://github.com/bazelbuild/intellij/blob/4c91d565062e152cace081bf3bfc8860008dc145/base/src/com/google/idea/blaze/base/vcs/git/GitBlazeVcsHandler.java#L53-L66) is how `GitBlazeVcsHandler` calculates the working set of files. The troubling part is how the [SHA is calculated in `getUpstreamSha`](https://github.com/bazelbuild/intellij/blob/4c91d565062e152cace081bf3bfc8860008dc145/base/src/com/google/idea/blaze/base/vcs/git/GitBlazeVcsHandler.java#L123-L141).

I propose to change the default for [`BlazeUserSettings#expandSyncToWorkingSet`](https://github.com/bazelbuild/intellij/blob/4c91d565062e152cace081bf3bfc8860008dc145/base/src/com/google/idea/blaze/base/settings/BlazeUserSettings.java#L72) from `true` to `false`. It seems this automatic expansion of the target set based on changed files can bring more harm than benefits for use cases similar to mine.

Unfortunately I wasn't able to find more information about motivation for `expandSyncToWorkingSet` feature because this feature predates the publicly available Git history.

**Comment by fkorotkov** [comment]
Opened https://github.com/bazelbuild/intellij/pull/3229

**Comment by alice-ks** [comment]
Disabling this functionality wouldn't work for everybody. I've heard that others in the community use the behavior of "Expand Sync to Working Set" as a workaround for pulling in dependencies which are not attached via the regular Blaze sync (due to gaps which might exist for their specific setup). I wouldn't want to break them.

Apart from this, "Expand Sync to Working Set" is a functionality which is used by our internal developers. We can't simply change the default behavior for them after so many years.

You're welcome to improve the implementation of `GitBlazeVcsHandler`.

An alternative could be to find out how the plugin's settings could be made available within the default IDE settings (reachable via "File > New Projects Setup > Settings for New Projects") and whether such settings could be easily provided/adjusted from outside of the original plugin.

**Comment by fkorotkov** [comment]
My main point is that it seems for me that this option should've been disabled by default in the first place. And people that want this "workaround for pulling in dependencies which are not attached via the regular Blaze sync" should explicitly enable it.

I think the question here is what is the right behaviour for the majority of people. I'm convinced that the option should be disabled by default because it causes frustrations that I described.

What if we'll disable it only for newly imported projects so it won't change behaviour for existing projects?
